### PR TITLE
log_backup: move all download request into the same runtime

### LINKED ISSUE #14285: PITR: dispatch task is gone: runtime dropped the dispatch task
### What version of TiKV are you using?

nightly

### Steps to reproduce

restore 24h log data to a large cluster from s3 storage(minio) with cache id enabled. https://github.com/pingcap/tidb/pull/41305

### What did you expect?
no error `dispatch task is gone: runtime dropped the dispatch task` in br.log even restore success.

### What did happened?
image

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed and how it works?

Issue Number: Close #14285

What's Changed:
Because the implementation of `hyper` (see https://github.com/hyperium/hyper/issues/2112), the lifetime of the client must be longer than the runtime.
However, because of some historic reasons, we are using a temporary runtime for S3 requests(!). Which may lead to the `runtime dropped` error (again!).

```commit-message
Download tasks will executed in a tiny runtime for now.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Manual test
(Generally this problem can only be tested by an end-to-end test, because the problem is S3 specified...)
The test is trivial: just run a PITR, and check whether there are "dispatch error".
#### BEFORE(Current release-6.5):
image

#### AFTER(This PR):
image

### Release note 

```release-note
Fixed a bug that may cause restoring from S3 get slow.
```

### CODE REVIEW

File: components/sst_importer/src/sst_importer.rs [hunk 1]
```diff
@@ -291,7 +290,20 @@ impl SstImporter {
...
         let switcher = ImportModeSwitcher::new(cfg);
         let cached_storage = CacheMap::default();
-        let download_rt = tokio::runtime::Builder::new_current_thread()
+        // We are going to run some background tasks here, (hyper needs to maintain the
+        // connection, the cache map needs gc intervally.) so we must create a
+        // multi-thread runtime, given there isn't blocking, a single thread runtime is
+        // enough.
+        let download_rt = tokio::runtime::Builder::new_multi_thread()
+            .worker_threads(1)
+            .thread_name("sst_import_tiny_runtime")
```
Comment (overvenus) [inline components/sst_importer/src/sst_importer.rs:299]: What does "tiny" means?

File: components/sst_importer/src/sst_importer.rs [hunk 2]
```diff
@@ -291,7 +290,20 @@ impl SstImporter {
...
+            .after_start_wrapper(|| {
+                tikv_alloc::add_thread_memory_accessor();
+                file_system::set_io_type(IoType::Import);
```
Comment (overvenus) [inline components/sst_importer/src/sst_importer.rs:302]: Could you add some comments about setting io type?

File: components/sst_importer/src/sst_importer.rs (hunk 1 above)
Comment (YuJuncen) [reply components/sst_importer/src/sst_importer.rs:299]: I guess perhaps we can rename it "misc_worker" or something?

**Review by overvenus — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]

**Pull request merged**